assets/pages/new-contributions-landing/contributionsLandingActions.js

```diff
@@ -198,12 +207,17 @@ const onThirdPartyPaymentAuthorised = (paymentAuthorisation: PaymentAuthorisatio
 
     switch (state.page.form.contributionType) {
       case 'ONE_OFF':
-        dispatch(executeOneOffPayment(makeOneOffPaymentData(paymentAuthorisation, state)));
+        // Why no mention of PayPal?
+        // Executing a one-off PayPal payment happens on the backend in the /paypal/rest/return
+        // endpoint, after PayPal redirects the browser back to our site.
+        if (state.page.form.paymentMethod === 'Stripe') {
+          dispatch(executeStripeOneOffPayment(makeStripeOneOffPaymentData(paymentAuthorisation, state)));
+        }
         return;
 
       case 'ANNUAL':
       case 'MONTHLY':
-        dispatch(setupRegularPayment(makeRegularPaymentData(paymentAuthorisation, state)));
+        dispatch(setupRegularPayment(makeRegularPaymentRequest(paymentAuthorisation, state)));
```

Comment:
Do we need to add `if (state.page.form.paymentMethod === 'Stripe') {` before this like we do with the above? Or is it payment method agnostic?

Reply:
it's payment method agnostic, so far - it handles both Stripe and Direct Debit with the same logic